### PR TITLE
Refactor KIR Step 7 - Remove kir dispatch

### PR DESCRIPTION
**Pull request by csarofeen**
This was pretty straightforward, adding kir nodes to cuda/dispatch.(h/cpp) and removing kernel_ir_dispatch.(h/cpp).

Also removes: `kirDispatch` and `kirConstDispatch`
Adds place holders in mutator for Kir nodes.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/mutator.cpp [hunk 1]
```diff
@@ -13,6 +13,22 @@ namespace cuda {
 
 // MUTATE FUNCTIONS FOR VALS
 
+Statement* OptOutMutator::mutate(Bool* b) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/mutator.cpp:16]: Just moved these to be in a consistent ordering.

File: torch/csrc/jit/codegen/cuda/mutator.cpp [hunk 2]
```diff
@@ -214,6 +194,33 @@ Statement* OptOutMutator::mutate(BroadcastOp* bop) {
   return bop;
 }
 
+Statement* OptOutMutator::mutate(Split* s) {
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/mutator.cpp:197]: Just moved Split/Merge to be in a consistent ordering

**Comment by naoyam** [comment]
Just to remind https://github.com/csarofeen/pytorch/pull/1347#discussion_r779111234

**Comment by naoyam** [comment]
> Just to remind [#1347 (comment)](https://github.com/csarofeen/pytorch/pull/1347#discussion_r779111234)

Looks good other than the above issue.

**Comment by naoyam** [comment]
I just noticed clang gives lots of warnings:

```
In file included from ../torch/csrc/jit/codegen/cuda/ir_printer.h:5:
../torch/csrc/jit/codegen/cuda/ir_iostream.h:51:16: warning: 'torch::jit::fuser::cuda::IrPrinter::handle' hides overloaded virtual functions [-Woverloaded-virtual]
  virtual void handle(Fusion& f) {
               ^
../torch/csrc/jit/codegen/cuda/dispatch.h:120:16: note: hidden overloaded virtual function 'torch::jit::fuser::cuda::OptOutConstDispatch::handle' declared here: type mismatch at 1st parameter ('const kir::Predicate *' vs 'torch::jit::fuser::cuda::Fusion &')
  virtual void handle(const kir::Predicate*);
```

Inserting `using OptInConstDispatch::handle;` in the `IrPrinter` class removes these warnings. 

The warnings may be found with the prior steps of the RPs.

**Comment by naoyam** [comment]
> I just noticed clang gives lots of warnings:
> 
> ```
> In file included from ../torch/csrc/jit/codegen/cuda/ir_printer.h:5:
> ../torch/csrc/jit/codegen/cuda/ir_iostream.h:51:16: warning: 'torch::jit::fuser::cuda::IrPrinter::handle' hides overloaded virtual functions [-Woverloaded-virtual]
>   virtual void handle(Fusion& f) {
>                ^
> ../torch/csrc/jit/codegen/cuda/dispatch.h:120:16: note: hidden overloaded virtual function 'torch::jit::fuser::cuda::OptOutConstDispatch::handle' declared here: type mismatch at 1st parameter ('const kir::Predicate *' vs 'torch::jit::fuser::cuda::Fusion &')
>   virtual void handle(const kir::Predicate*);
> ```
> 
> Inserting `using OptInConstDispatch::handle;` in the `IrPrinter` class removes these warnings.
> 
> The warnings may be found with the prior steps of the RPs.

Just noticed it's resolved in the step 10 PR, so it should be fine as is.

**Pull request merged**